openmc/filter.py

```diff
@@ -0,0 +1,1810 @@
+        # This chunk of code handles this docstring inheritance manually so that
+        # the autodocumentation will pick it up.
+        if name != 'Filter':
+            # Look for newly-defined functions that were also in Filter.
+            for func_name in namespace:
+                if func_name in Filter.__dict__:
+                    # Inherit the docstring from Filter if not defined.
+                    if isinstance(namespace[func_name],
+                                  (classmethod, staticmethod)):
+                        new_doc = namespace[func_name].__func__.__doc__
+                        old_doc = Filter.__dict__[func_name].__func__.__doc__
+                        if new_doc is None and old_doc is not None:
+                            namespace[func_name].__func__.__doc__ = old_doc
+                    else:
+                        new_doc = namespace[func_name].__doc__
+                        old_doc = Filter.__dict__[func_name].__doc__
+                        if new_doc is None and old_doc is not None:
+                            namespace[func_name].__doc__ = old_doc
+
+        # Make the class.
+        return super().__new__(cls, name, bases, namespace, **kwargs)
+
+
+class Filter(IDManagerMixin, metaclass=FilterMeta):
+    """Tally modifier that describes phase-space and other characteristics.
+
+    Parameters
+    ----------
+    bins : Integral or Iterable of Integral or Iterable of Real
+        The bins for the filter. This takes on different meaning for different
+        filters. See the docstrings for sublcasses of this filter or the online
+        documentation for more details.
+    filter_id : int
+        Unique identifier for the filter
+
+    Attributes
+    ----------
+    bins : Integral or Iterable of Integral or Iterable of Real
+        The bins for the filter
+    id : int
+        Unique identifier for the filter
+    num_bins : Integral
+        The number of filter bins
+
+    """
+
+    next_id = 1
+    used_ids = set()
+
+    def __init__(self, bins, filter_id=None):
+        self.bins = bins
+        self.id = filter_id
+
+    def __eq__(self, other):
+        if type(self) is not type(other):
+            return False
+        elif len(self.bins) != len(other.bins):
+            return False
+        else:
+            return np.allclose(self.bins, other.bins)
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __gt__(self, other):
+        if type(self) is not type(other):
+            if self.short_name in _FILTER_TYPES and \
+                other.short_name in _FILTER_TYPES:
+                delta = _FILTER_TYPES.index(self.short_name) - \
+                        _FILTER_TYPES.index(other.short_name)
+                return delta > 0
+            else:
+                return False
+        else:
+            return max(self.bins) > max(other.bins)
+
+    def __lt__(self, other):
```

Comment:
Seems odd that the `gt` implementation is much more complex than the `lt` implementation. Is it possible to either simplify `gt` to mimic `lt`? Or is there a business logic reason for having the operators behave differently